src/hooks/useCampPartygoers.ts

```diff
@@ -1,30 +1,49 @@
-import { useMemo } from "react";
+import { useCallback, useMemo } from "react";
 
 import { User } from "types/User";
 import { WithId } from "utils/id";
 import { partygoersSelector } from "utils/selectors";
 
 import { useConnectPartyGoers } from "./useConnectPartyGoers";
+import { useConnectRelatedVenues } from "./useConnectRelatedVenues";
 import { useSelector } from "./useSelector";
 import { useUserLastSeenThreshold } from "./useUserLastSeenThreshold";
+import { useVenueId } from "./useVenueId";
 
 /**
  * Hook to retrieve a list of partygoers who were in venueName
  * within lastSeenThreshold.
  *
  * @param venueName venueName to filter partygoers by
  */
-export const useCampPartygoers = (venueName: string): WithId<User>[] => {
+export const usePartygoersThreshold = (venueName: string): WithId<User>[] => {
   useConnectPartyGoers();
+  const venueId = useVenueId();
   const partygoers = useSelector(partygoersSelector) ?? [];
 
+  const { parentVenue, relatedVenues } = useConnectRelatedVenues({
+    venueId,
+    withEvents: true,
+  });
+
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  return useMemo(
-    () =>
-      partygoers.filter(
-        (partygoer) => partygoer?.lastSeenIn?.[venueName] > lastSeenThreshold
-      ),
-    [partygoers, lastSeenThreshold, venueName]
+  const parentVenueName = parentVenue?.name ?? "";
+
+  const filterPartygoersByHierarchy = useCallback(
+    (partygoer: WithId<User>) => {
+      return relatedVenues?.map((venue) => {
+        return (
+          partygoer?.lastSeenIn?.[parentVenueName || venueName || venue.name] >
+          lastSeenThreshold
+        );
```

Comment:
What is the purpose of `parentVenueName || venueName || venue.name` here? Do we only want to use the 'next name' if the previous was falsy? Or do we want to check all 3 to see if they are within the threshold?

Reply:
We need to check one by one, the partygoers should be checked in the heirarchy and that's why i'm checking for parentVenue if it doesn't exist then it checks for the current venue name.

Reply:
Not relevant anymore, I suppose